pkgs/by-name/gr/grub2-xen-images/package.nix

```
@@ -0,0 +1,65 @@
+{
+  lib,
+  stdenv,
+  buildPvImage ? false,
+  grub2_xen,
+  buildPvhImage ? false,
+  grub2_xen_pvh,
+}:
+
+assert (buildPvImage != buildPvhImage);
+let
+  efiSystemsBuild = {
+    i686-linux.target = "i386";
+    x86_64-linux.target = if buildPvhImage then "i386" else "x86_64";
+    armv7l-linux.target = "arm";
+    aarch64-linux.target = "aarch64";
+    riscv32-linux.target = "riscv32";
+    riscv64-linux.target = "riscv64";
+  };
+
+in
+(
+
+  stdenv.mkDerivation rec {
+    name = "grub2-xen-images";
+
+    configs = ./configs;
+
+    buildInputs = lib.optional buildPvImage grub2_xen ++ lib.optional buildPvhImage grub2_xen_pvh;
+
+    buildCommand =
+      let
+        grubVariant = if buildPvImage then "xen" else "xen_pvh";
+      in
+      ''
+        cp "${configs}"/* .
+        tar -cf memdisk.tar grub.cfg
+        # We include all modules except all_video.mod as otherwise grub will fail printing "no symbol table"
+        # if we include it.
+        grub-mkimage \
+          -O "${efiSystemsBuild.${stdenv.hostPlatform.system}.target}-${grubVariant}" \
+          -c grub-bootstrap.cfg \
+          -m memdisk.tar \
+          -o "grub-${efiSystemsBuild.${stdenv.hostPlatform.system}.target}-${grubVariant}.bin" \
```

```suggestion
          -o "grub-${imageName}.bin" \
```
Since `${efiSystemsBuild.${stdenv.hostPlatform.system}.target}-${grubVariant}` is repeated quite a few times, add it to the `let...in` expression above.

---

Oh yes, 4 times. I should have noticed this when reusing the old `grub2_pvgrub_image` package.


I think we should call it `grubFormat`, as it's seen in `grub-mkimage(1)` man page, this reflects how the upstream calls the "target_cpu-platform" argument.